be/src/exec/vectorized/join_hash_map.tpp

```diff
@@ -394,11 +394,28 @@ Status JoinHashMap<PT, BuildFunc, ProbeFunc>::_probe_output(ChunkPtr* probe_chun
 
     for (size_t i = 0; i < _table_items->probe_column_count; i++) {
         SlotDescriptor* slot = _table_items->probe_slots[i];
-        auto& column = (*probe_chunk)->get_column_by_slot_id(slot->id());
-        if (!column->is_nullable()) {
-            _copy_probe_column(&column, chunk, slot, to_nullable);
+        auto &column = (*probe_chunk)->get_column_by_slot_id(slot->id());
+
+        if (_table_items->output_slots.empty()
+                || std::find(_table_items->output_slots.begin(), _table_items->output_slots.end(), slot->id()) != _table_items->output_slots.end()
```

Comment:
When there are many columns, using Vector may have performance problems

Reply:
no need to do it every time, just do it once.

Reply:
Is it okay to use set instead?